lib/jobs/obm-control.js

```diff
@@ -64,18 +62,33 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
+            self.settings = _.find(node.obmSettings, function(setting) {
                 return setting ? setting.service === self.obmServiceName : false;
             });
 
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                // throw error if the exact obm service doesn't exist
```

Comment:
I think it's more reasonable to move line 65-67 here because you don't need to lookup obm-service if no explicit obmServiceName is specified.